ansible_rulebook/job_template_runner.py

```diff
@@ -0,0 +1,232 @@
+        if search:
+            job_template = await self._search_job_template(name, organization)
+            self.template_id_cache[key] = job_template["id"]
+        return job_template
+
+    async def get_job_template_by_id(self, id: int) -> dict:
+        async with aiohttp.ClientSession(
+            headers=self._auth_headers()
+        ) as session:
+            slug = f"{self.JOB_TEMPLATE_SLUG}/{id}/"
+            response = await self._get_page(session, slug, {})
+            return json.loads(response["body"])
+
+    async def _search_job_template(self, name: str, organization: str) -> dict:
+        job_template = await self._search_list(
+            self.JOB_TEMPLATE_SLUG,
+            {"name": name, "summary_fields.organization.name": organization},
+        )
+        if not job_template:
+            raise JobTemplateNotExistException(
+                "Job template %s in organization %s does not exist"
+                % (name, organization)
+            )
+
+        return job_template
+
+    async def _search_list(self, href_slug: str, query: dict) -> dict:
+        async with aiohttp.ClientSession(
+            headers=self._auth_headers()
+        ) as session:
+            url_info = urlparse(href_slug)
+            params = dict(parse_qsl(url_info.query))
+            while True:
+                response = await self._get_page(session, url_info.path, params)
+                json_body = json.loads(response["body"])
+                for obj in json_body["results"]:
+                    match = True
+                    for key, value in query.items():
+                        if dpath.get(obj, key, ".") != value:
+                            match = False
+                            break
+                    if match:
+                        return obj
+
+                if json_body.get("next", None):
+                    params["page"] = params.get("page", 1) + 1
+                else:
+                    break
+
+    async def _get_page(self, session, href_slug, params):
+        url = urljoin(self.host, href_slug)
+        async with session.get(url, params=params) as response:
+            response_text = dict(
+                status=response.status, body=await response.text()
+            )
+        if response_text["status"] != 200:
+            raise ControllerApiException(
+                "Failed to get from %s. Status: %s, Body: %s"
+                % (
+                    url,
+                    response_text["status"],
+                    response_text.get("body", "empty"),
+                )
+            )
+        return response_text
+
+    def _auth_headers(self):
+        return dict(Authorization=f"Bearer {self.token}")
+
+    async def run_job_template(
+        self,
+        name: str,
+        organization: str,
+        job_params: dict,
+        event_handler: Callable[[dict], Any],
+    ) -> str:
+        job_template = await self.get_job_template(name, organization)
```

Comment:
Do you know that you can call the `job_template` by the name instead by the id? I think it could make simpler the code. 

Ref: https://www.linkedin.com/pulse/calling-ansible-automation-platform-api-using-job-names-griffiths/

Reply:
cool. will try it out.